### DISCUSSION TITLE
Electron IPC Messages support

### DISCUSSION THREAD

**Discussion by angelfraga** [trpc/trpc#2340] [Q&A] [answered]
Hi, first of all thanks for that great work! 

Some context:

In last days I've being working at work in order to align a client UI with the electron IPCMessage handlers typing.

While I think I've got a good progress using a shared library, in some point I've faced with that library and I asked myself if it would be possible to use [trpc](https://github.com/trpc/trpc) for strongly typing IPC communication https://www.electronjs.org/docs/latest/api/ipc-main .

I think it would have some similarities with Websocket implementation https://trpc.io/docs/subscriptions 

What do you think? 
Would it make sense giving that support ? 
Can we use an already existing feature for that purpose ?

**Comment by jsonnull** [comment, marked as answer]
Hey, I created the package [electron-trpc](https://github.com/jsonnull/electron-trpc) recently to do exactly this! If you're interested in trying it, feel free to get in touch with me and I can help out as needed.

I'm personally not sure that the integration belongs in official tRPC packages, but I have been in contact with some of the maintainers recently about ensuring the right public tRPC APIs are in place to allow these kind of integrations to be built by the community.

**Reply by angelfraga** [reply]
Thanks, that looks promising 👍  I will have a look soon as possible.

**Reply by jsonnull** [reply]
FYI, I in the last few minutes I updated the docs and published version 0.0.3, which does a small API simplification. Hopefully everything is clear and easy to follow, do let me know if you run into any issues.

**Reply by waptik** [reply]
> FYI, I in the last few minutes I updated the docs and published version 0.0.3, which does a small API simplification. Hopefully everything is clear and easy to follow, do let me know if you run into any issues.

Hey, I just want to say that it'd be great if you created a basic electron app using your package to show its usage, then add the repo link in the readme of electron-trpc. This will help devs really appreciate it.
Just my 2 cents.

**Reply by jsonnull** [reply]
Sure thing @waptik, I'll add that today
